### ISSUE TITLE
Bug in num_to_link ?

### ISSUE THREAD

**Issue by mrkiwi-nz** [django-helpdesk/django-helpdesk#511]
**What happens;**

```
num_to_link('this is ticket#999')
u"this is ticke <a href='/helpdesk/tickets/999/' class='ticket_link_status ticket_link_status_Open'>#999</a>"

```
**What I expected to happen**
```
num_to_link('this is ticket#999')
u"this is ticket <a href='/helpdesk/tickets/999/' class='ticket_link_status ticket_link_status_Open'>#999</a>"

```
**Why:**
`regex = r"(?:[^&]|\b|^)#(\d+)\b"`
So ... (non-capturing) match a character that is: not "&" OR is a word boundary OR is the start of the line
Then, find a "#", then capture digits until you hit a word boundary

Have I read this right? If so, what is the purpose of the non-capturing 'not a "&" ' part? It swallows the last character before the #, but was it meant to not capture a string like "&#999" ?

**Comment by gwasser** [comment]
I suspect the `&` part is so we don't interpret HTML entities as ticket numbers. I'm sure that regex could be written a little clearer but I'd have to think about what rules we want to use. I'll address this together with #512 and #513.

**Comment by gwasser** [comment]
This fix was merged in as part of #532, so I'll close the ticket but reopen or file a new one if there's still trouble. Thanks!
